### PR TITLE
Fix MarkDown in TSDoc comments

### PR DESCRIPTION
**Pull request by tukusejssirs**
Changes:

- add blank line:
   - before lists;
   - between two lines of text which seem to be two separate paragraph (e.g. [`index.d.ts` (L290-291)](https://github.com/sindresorhus/execa/blob/main/index.d.ts#L290-L291));
- remove indentation before first-level in lists;
- specify syntax highlighting language as `typescript` for code blocks when missing or when `js` was used.

### CODE REVIEW

**Comment by tukusejssirs** [comment]
### Why this is needed?

VS Code outputs TSDoc comments correctly even without these changes, however Sublime Text (using LSP + TypeScript server, using [`sublime-markdown-popups`](https://github.com/facelessuser/sublime-markdown-popups)) does not output the HTML (converted from MarkDown) correctly.

It might be considered a bug in `sublime-markdown-popups`, however, I think that TSDoc comments _should_ have leading asterisk on each comment line.

Note: It might be good idea to add missing TSDocs for `ExecaSyncReturnValue` ([`index.d.ts` (L340)](https://github.com/sindresorhus/execa/blob/main/index.d.ts#L340)).

**Comment by sindresorhus** [comment]
> It might be considered a bug in sublime-markdown-popups, however, I think that TSDoc comments should have leading asterisk on each comment line.

You should report that to that project. I prefer without leading asterisk and will not change that.

**Comment by tukusejssirs** [comment]
Okay, thanks for your reply. "hat about the other changes I suggest

> Changes:
>
> - ~~fix TSDoc comments by adding leading asterisks;~~
> - add blank line:
>    - before lists;
>    - between two lines of text which seem to be two separate paragraph (e.g. [`index.d.ts` (L290-291)](https://github.com/sindresorhus/execa/blob/main/index.d.ts#L290-L291));
> - remove indentation before first-level in lists;
> - specify syntax highlighting language as `typescript` for code blocks when missing or when `js` was used.

**Comment by tukusejssirs** [comment]
I have removed the leading asterisks from TSDoc comments. I’ll squash the commits later.

Are the current changes acceptable to you? :smiley:

File: index.d.ts [hunk 1]
```diff
@@ -137,8 +142,11 @@ export interface CommonOptions<EncodingType> {
 
 	/**
 	Specify the kind of serialization used for sending messages between processes when using the `stdio: 'ipc'` option or `execaNode()`:
-		- `json`: Uses `JSON.stringify()` and `JSON.parse()`.
-		- `advanced`: Uses [`v8.serialize()`](https://nodejs.org/api/v8.html#v8_v8_serialize_value)
+
+	- `json`: Uses `JSON.stringify()` and `JSON.parse()`.
+	- `advanced`: Uses [`v8.serialize()`](https://nodejs.org/api/v8.html#v8_v8_serialize_value)
+
+	Requires Node.js `13.2.0` or later.
```
Comment (sindresorhus) [inline index.d.ts:149]: This is moot. Execa requires Node.js 14.

File: index.d.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ export type StdioOption =
 export interface CommonOptions<EncodingType> {
 	/**
 	Kill the spawned process when the parent process exits unless either:
-		- the spawned process is [`detached`](https://nodejs.org/api/child_process.html#child_process_options_detached)
-		- the parent process is terminated abruptly, for example, with `SIGKILL` as opposed to `SIGTERM` or a normal exit
+
```
Comment (sindresorhus) [inline index.d.ts:18]: I don't really see the point in having an empty line before the list?

File: index.d.ts [hunk 3]
```diff
@@ -460,7 +474,7 @@ Think of this as a mix of `child_process.execFile` and `child_process.spawn`.
 @returns A [`child_process` instance](https://nodejs.org/api/child_process.html#child_process_class_childprocess), which is enhanced to also be a `Promise` for a result `Object` with `stdout` and `stderr` properties.
 
 @example
-```
+```typescript
```
Comment (sindresorhus) [inline index.d.ts:477]: TypeScript is inferred. So you don't need any specifier.

File: index.d.ts [hunk 4]
```diff
@@ -472,14 +486,14 @@ console.log(stdout);
 const subprocess = execa('node');
 
 setTimeout(() => {
-	subprocess.cancel()
+  subprocess.cancel()
```
Comment (sindresorhus) [inline index.d.ts:489]: Don't make unrelated changes. It should use tab indentation.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 4 above)
Comment (tukusejssirs) [reply index.d.ts]: Fix in latest commit.

**Review by tukusejssirs — COMMENTED** [review]
I fixed most issues raised and for the rest I tried to provide a valid reason why I need it formatted like I propose, which while not required per se, it is optional in GitHub Flavoured Markdown as well as CommonMark.

I hope you’ll understand, @sindresorhus. :pray:

**Comment by sindresorhus** [comment]
I have to be honest, I'm not really interested in working around `sublime-markdown-popups` limitations. CommonMark is what I'm used to and prefer.

**Comment by sindresorhus** [comment]
> It is inferred in VS Code, but not in ST. When it is not specified, it’ll be output in single colour (no syntax highlighting), see below.

Seems like you should open an issue on ST or the plugin.
